src/model_parsers/TTAParser.cpp

```diff
@@ -198,35 +230,176 @@ std::optional<TTAIR_t::Symbol> TTAParser::ParseSymbolDeclaration(const std::stri
+                .currentLocation           = { comp.initialLocation.isImmediate, comp.initialLocation.identifier },
                 .isMain                    = comp.isMain,
-                .edges                     = ConvertEdgeListToEdgeMap(comp.edges),
-                .symbols                   = ConvertSymbolListToSymbolMap(comp.symbols)
+                .edges                     = ConvertEdgeListToEdgeMap(comp.edges, tta.symbols, comp.name),
         };
     }
     return tta;
 }
 
-std::unordered_map<std::string, TTASymbolType>
-TTAParser::ConvertSymbolListToSymbolMap(const std::vector<TTAIR_t::Symbol> &symbolList) {
-    std::unordered_map<std::string, TTASymbolType> map{};
-    std::for_each(symbolList.begin(), symbolList.end(), [&map] (auto& symbol) { map[symbol.identifier]=symbol.value; });
+TTA::SymbolMap TTAParser::ConvertSymbolListToSymbolMap(const std::vector<TTAIR_t::Symbol> &symbolList) {
+    TTA::SymbolMap map{};
+    std::for_each(symbolList.begin(), symbolList.end(), [&map] (auto& symbol) {
+        std::visit(overload(
+                [&](const int& value)            { map[symbol.identifier] = value; },
+                [&](const float& value)          { map[symbol.identifier] = value; },
+                [&](const bool& value)           { map[symbol.identifier] = value; },
+                [&](const TTATimerSymbol& value) { map[symbol.identifier] = value.current_value; }, // Timers are just read-only floats in the Tick-semantics
+                [&](const std::string& value)    { map[symbol.identifier] = value; }
+                ), symbol.value);
+    });
     return map;
 }
 
-std::unordered_map<std::string, TTA::Edge>
-TTAParser::ConvertEdgeListToEdgeMap(const std::vector<TTAIR_t::Edge> &edgeList) {
-    std::unordered_map<std::string, TTA::Edge> map{};
+#include <extensions/cparse_extensions.h> // TODO: Remove this. This is only used for debugging purposes atm
+std::unordered_multimap<std::string, TTA::Edge>
+TTAParser::ConvertEdgeListToEdgeMap(const std::vector<TTAIR_t::Edge> &edgeList, const TTA::SymbolMap& symbolMap, const std::string& debugCompName) {
+    std::unordered_multimap<std::string, TTA::Edge> edgeMap{};
+    calculator calc;
     for(auto& edge : edgeList) {
-        map[edge.sourceLocation.identifier] = {
-                .sourceLocation           = {edge.sourceLocation.identifier, edge.sourceLocation.isImmediate},
-                .targetLocation           = {edge.targetLocation.identifier, edge.targetLocation.isImmediate},
-                .guardExpression          = edge.guardExpression,   // TODO: Compile this as a thing
-                .updateExpression         = edge.updateExpression  // TODO: Compile this as a thing
-        };
+        // Compile the expressions (guard and update)
+        try {
+            if (!edge.guardExpression.empty()) {
+                calc.compile(edge.guardExpression.c_str(), symbolMap);
+                auto type = calc.eval()->type;
```

Comment:
There might be an issue with using `eval()` here. We dont want any side-effects just yet. Dig through cparse to see if you can just get the overall type of the expression without evaluating it.